clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -5077,6 +5077,21 @@ static void handleSYCLDeviceIndirectlyCallableAttr(Sema &S, Decl *D,
   handleSimpleAttribute<SYCLDeviceIndirectlyCallableAttr>(S, D, AL);
 }
 
+static void handleSYCLGlobalVarAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
+  if (!S.Context.getSourceManager().isInSystemHeader(D->getLocation())) {
+    S.Diag(AL.getLoc(), diag::err_attribute_wrong_decl_type_str)
+        << AL << "system header global variables";
+  }
+
+  const auto *VD = cast<VarDecl>(D);
+  if (VD && VD->isLocalVarDecl()) {
+    S.Diag(AL.getLoc(), diag::err_attribute_wrong_decl_type_str)
+        << AL << "global variables";
+  }
```

Comment:
This logic should live in Attr.td for the subject list, once we nail down what the predicate actually should be.